documentation/proposals/Proposal - 3.0 & 3.X Software Development Plan.md

```diff
@@ -0,0 +1,149 @@
+- Bugfixes from 3.0 Preview 3
+- Windowing integrations for WPF and WinForms have been developed and have at least basic OpenGL support. The support may not be the most high performance possible at this time.
+- SIMD APIs in Maths have been complete, and work has started to integrate them into the other Maths types in the most common cases.
+- Ample work has been done to migrate 2.0 code to 3.0 code to evaluate differences in public API, fixing them where we deem necessary.
+
+3.0 Preview 4 is a production-ready preview and users are encouraged to start integrating this preview into their code. This preview should aim to release in late October.
+
+### 3.0 Preview 5
+
+This is the last preview and is primarily a bugfix release. All breaking changes should've been done in previous previews, but if this is not the case all forseeable breaking must be 100% done in this preview. In this version:
+- Bugfixes from 3.0 Preview 4
+- A windowing integration for MAUI has been developed and has at least basic OpenGL(ES) support in a state that is as high-performance and as smoothed-out as possible.
+- If time permits, a windowing integration for Avalonia has been developed and has at least basic OpenGL(ES) support. If there is not enough time, this can be pushed to 3.X.
+- SIMD APIs should be integrated into Maths in as many common cases as possible. Ongoing performance improvements may be done in 3.X. This preview should aim to release in early November.
+
+## Problems identified in past development
+
+- We have severely lacking documentation
+    - The intention is that all developers of large amounts of code write implementation documentation and/or "orientation guides" for their codebases informing readers of all major things there is to know in their code.
+    - We should also write documentation containing examples on using as many features of the surface APIs as possible \[for high level utilities\]
+    - We will enforce XML documentation in all manually-written utilities and as much as possible in bindings.
+    - If time permits, we should productionize our website powered by Raisin.
+- There's not a lot of planning
+    - We have solved this in the form of the proposal you are reading and all linked proposals: getting all the design done now and documented now, to prevent design debates later down the line. This should reduce friction when actually working on the library.
+    - We have been keeping the working group and key stakeholders in the loop with the 3.0 kickoff (again, see this proposal you are reading)
+    - The team are trying to communicate with eachother and figure out how to distribute work among themselves depending on individual circumstances and free time
+        - Codeowners have been established
+- Barrier to entry for external contributors is very high
+    - Documentation should help with this.
+    - We should at least consider introducing something like stylecop to ensure code is readable and easy to navigate.
```

Comment:
.NET 5 and up ship with enhanced warnings. Also nullability analysis everywhere please.